src/Jackalope/Transport/DoctrineDBAL/Client.php

```diff
@@ -2201,6 +2202,11 @@ public function getBinaryStream($path)
             array($nodeId, $propertyName, $this->workspaceName)
         );
 
+        if (count($data) === 0) {
+            throw new RepositoryException('No binary data found in stream');
```

Comment:
i think the bug is further down, on line 2228 where we assume that there is exactly one stream when there are not > 1.

or is this only a problem when people copied nodes before your bugfix and now have an incomplete/inconsistent database?
i think the sanity check has to happen in Property::getBinary. that method has the `length` field of the property which in case of multivalue is an array and can thus see when it did not get the number of binary from storage that was to be expected.

does that make sense?